.github/workflows/ci.yml

```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 13.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: yarn install --frozen-lockfile
+    - run: yarn format
+    - run: yarn test:unit
+    - run: yarn test:int
```

Comment:
I could be wrong, but shouldn't just `yarn test` work? It should recursively run everything in the `test` directory, and we don't want it doing things like recording _new_ Nock VCRs in this environment anyway.

Reply:
Hmmm,

Let me see. `yarn test:int` will run `NOCK_BACK_MODE=record mocha --recursive ./test/integration --timeout 10000`

To review what that means:

* `record`: use recorded nocks, record new nocks

Other options:

* `wild`: all requests go out to the internet, don't replay anything, doesn't record anything
* `dryrun`: The default, use recorded nocks, allow http calls, doesn't record anything, useful for writing new tests
* `record`: use recorded nocks, record new nocks
* `lockdown`: use recorded nocks, disables all http calls even when not nocked, doesn't record

Looking at these definitions, I think you're right. `record` is what we want when developing locally (it will use recorded tests if they're there, otherwise record new ones)

For CI the default makes the most sense. Maybe in test we want to run `NOCK_BACK_MODE=lockdown yarn test`? That's the same idea you had, except setting the Nock mode to lockdown so that we don't accidentally make unintended http requests.

Reply:
I can get behind that. The default (dryrun) _shouldn't_ make any live requests since the only requests should be `nocked` anyway, but it's not the worst idea just in case.

Reply:
I had the same thought... okay, committing!